libraries/AP_Logger/AP_Logger_MAVLink.cpp

```diff
@@ -565,27 +600,17 @@ bool AP_Logger_MAVLink::send_log_block(struct dm_block &block)
     }
     // don't completely fill buffers - and also ensure there's enough
     // room to send at least one packet:
-    const uint16_t min_payload_space = 500;
-    static_assert(MAVLINK_MSG_ID_REMOTE_LOG_DATA_BLOCK_LEN <= min_payload_space,
-                  "minimum allocated space is less than payload length");
-    if (_link->txspace() < min_payload_space) {
+    if (_link->txspace() < MAVLINK_MSG_ID_REMOTE_LOG_DATA_BLOCK_LEN) {
```

Comment:
This code wouldn't have been put in here for fun - it is definitely leaving extra space for other messages to go out in the buffer.